### PR TITLE
docs: update evals docs

### LINKED ISSUE #1391: [Docs] Evals documentation missing some crucial context / information.
**Is your report related to a problem? Please describe.**
There are a few things that are glossed over the evaluation documentation (Evaluation and Writing an Evaluator Plugin)

Evaluator:
 - How to pass `auth` to the `eval:flow` command.
 - The format of the `input.json` file for `eval:flow`.
 
Writing an Evaluator Plugin:
 - Add more context to code snippets (eg: it is unclear if `regexMatchScore` and `fillScores` are provided in Genkit or if the user should implement)
 - Provide a simple explanation of the nature of `EvaluatorFn`.

**Additional context**
https://discord.com/channels/1255578482214305893/1309828408141807657

Some Genkit imports were missing / not updated to `0.9`. This was addressed in this PR by @DenisVCode: https://github.com/firebase/genkit/pull/1388

### PR DESCRIPTION
**Pull request by ssbushi**
Fix #1391 

Checklist (if applicable):
- [x] Docs updated

### CODE REVIEW

File: docs/evaluation.md
Comment (chrisraygill) [inline docs/evaluation.md:1]: On line 261:

```ts
filePath = path.resolve(filePath);
```

Does `path` need to be imported?

Comment (chrisraygill) [inline docs/evaluation.md:1]: On line 262:

```ts
const pdfTxt = await run("extract-text", () => extractText(filePath));
```

Does `exractText` need to be imported?

Comment (chrisraygill) [inline docs/evaluation.md:1]: On line 265:

```ts
chunk(pdfTxt, chunkingConfig)
```

Where does `chunkingConfig` come from?

**Review by chrisraygill — CHANGES_REQUESTED** [review]
In samples, it's unclear where the `ai` instance, `z`, and potentially other imports are coming from.

**Review by chrisraygill — COMMENTED** [review]
on line 128, should `defineEvaluator` be `ai.defineEvaluator` ?

File: docs/plugin-authoring-evaluator.md [hunk 1]
```diff
@@ -140,6 +143,47 @@ export function createDeliciousnessEvaluator<
 }
 ```
 
+The `defineEvaluator` method is similar to other Genkit constructors like `defineFlow`, `defineRetriever` etc. The user should provide an `EvaluatorFn` to the `defineEvaluator` callback. The `EvaluatorFn` accepts a `BaseEvalDataPoint` which corresponds to a single entry in a dataset under evaluation, along with an optional custom options parameter if specified. The function, should process the datapoint and return an `EvalResponse` object. 
+
+Here are the Zod Schemas for `BaseEvalDataPoint` and `EvalResponse`:
+
+```ts
+export const BaseEvalDataPoint = z.object({
+  testCaseId: z.string(),
```
Comment (chrisraygill) [inline docs/plugin-authoring-evaluator.md:152]: `testCaseId` is duplicated in the schema

**Review by chrisraygill — APPROVED** [review]

**Comment by pavelgj** [comment]
please follow https://www.conventionalcommits.org/en/v1.0.0/ for PRs

File: docs/evaluation.md
Comment (ssbushi) [reply docs/evaluation.md:1]: It's a custom method, we cover this in our RAG documentation. I've added a comment here. LMK if that is not sufficient.

**Pull request merged**